pkg/asset/machines/gcp/gcpmachines.go

```diff
@@ -0,0 +1,183 @@
+	bootstrapCapiMachine := createCAPIMachine(bootstrapGCPMachine.Name, dataSecret, infraID)
+
+	result = append(result, &asset.RuntimeFile{
+		File:   asset.File{Filename: fmt.Sprintf("10_machine_%s.yaml", bootstrapCapiMachine.Name)},
+		Object: bootstrapCapiMachine,
+	})
+	return result, nil
+}
+
+// Create a CAPG-specific machine
+func createGCPMachine(name string, installConfig *installconfig.InstallConfig, infraID string, mpool *gcptypes.MachinePool, imageName string) *capg.GCPMachine {
+
+	// Use the rhcosImage as image name if not defined
+	var osImage string
+	if mpool.OSImage == nil {
+		osImage = imageName
+	} else {
+		osImage = mpool.OSImage.Name
+	}
+
+	// TODO tags aren't currently being set in GCPMachine which only has
+	// AdditionalNetworkTags []string
+
+	masterSubnet := installConfig.Config.Platform.GCP.ControlPlaneSubnet
+	if masterSubnet == "" {
+		masterSubnet = gcptypes.DefaultSubnetName(infraID, "master")
+	}
+
+	gcpMachine := &capg.GCPMachine{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
+			Kind:       "GCPMachine",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				"cluster.x-k8s.io/control-plane": "",
+			},
+		},
+		Spec: capg.GCPMachineSpec{
+			InstanceType:     mpool.InstanceType,
+			Subnet:           ptr.To(masterSubnet),
+			AdditionalLabels: getLabelsFromInstallConfig(installConfig, infraID),
+			Image:            ptr.To(osImage),
+			RootDeviceType:   ptr.To(capg.DiskType(mpool.OSDisk.DiskType)),
+			RootDeviceSize:   mpool.OSDisk.DiskSizeGB,
+		},
+	}
+	// Set optional values from machinepool
+	if mpool.OnHostMaintenance != "" {
+		gcpMachine.Spec.OnHostMaintenance = ptr.To(capg.HostMaintenancePolicy(mpool.OnHostMaintenance))
+	}
+	if mpool.ConfidentialCompute != "" {
+		gcpMachine.Spec.ConfidentialCompute = ptr.To(capg.ConfidentialComputePolicy(mpool.ConfidentialCompute))
+	}
+	if mpool.SecureBoot != "" {
+		shieldedInstanceConfig := capg.GCPShieldedInstanceConfig{}
+		shieldedInstanceConfig.SecureBoot = capg.SecureBootPolicyEnabled
+		gcpMachine.Spec.ShieldedInstanceConfig = ptr.To(shieldedInstanceConfig)
+	}
+	if mpool.ServiceAccount != "" {
+		serviceAccount := &capg.ServiceAccount{
+			Email: mpool.ServiceAccount,
+			// Set scopes to value defined at
+			// https://cloud.google.com/compute/docs/access/service-accounts#scopes_best_practice
+			Scopes: []string{"https://www.googleapis.com/auth/cloud-platform"},
+		}
+		gcpMachine.Spec.ServiceAccount = serviceAccount
+	}
+
+	return gcpMachine
+}
+
+// Create a CAPI machine based on the CAPG machine
+func createCAPIMachine(name string, dataSecret string, infraID string) *capi.Machine {
+	machine := &capi.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				"cluster.x-k8s.io/control-plane": "",
+			},
+		},
+		Spec: capi.MachineSpec{
+			ClusterName: infraID,
+			// Leave emotry until ignition support is added
```

Comment:
nit? I'm guessing you meant empty.
```suggestion
			// Leave empty until ignition support is added
```

Reply:
Yep, fixed.